src/Rules/Comparison/MatchExpressionRule.php

```diff
@@ -44,12 +44,13 @@ public function processNode(Node $node, Scope $scope): array
 	{
 		$matchCondition = $node->getCondition();
 		$matchConditionType = $scope->getType($matchCondition);
-		$nextArmIsDead = false;
+		$nextArmIsDeadForType = false;
+		$nextArmIsDeadForNativeType = false;
 		$errors = [];
 		$armsCount = count($node->getArms());
 		$hasDefault = false;
 		foreach ($node->getArms() as $i => $arm) {
-			if ($nextArmIsDead) {
+			if ($nextArmIsDeadForNativeType || $nextArmIsDeadForType && $this->treatPhpDocTypesAsCertain) {
```

Comment:
Please use parentheses to disambiguate the condition. And please rebase on top of 1.10.x so we can see whether there are new failures. Thanks.

Reply:
It's rebased with parentheses added